openshift/fluent.conf

```diff
@@ -49,25 +49,36 @@
   capitalize_regex_backreference yes
   <rule>
     key     code
-    pattern ^4\d\d$
+    # filter for 5** HTTP response codes
+    pattern ^5\d\d$
     tag     ${tag}.error
   </rule>
+
+  <rule>
+    key     log
+    # filter logs containing string 'EHOSTUNREACH'. for example: 'ERR! Error: connect EHOSTUNREACH 172.50.242.155:6379'
+    pattern EHOSTUNREACH
+    tag     ${tag}.error
+  </rule>
+
 </match>
 
-# reformat dogs_node.error logs for output via notification
+
+# reformat *.error logs into a Discord post
+# content ${record["product"]} - HTTP response status code ${record["code"]}, hostname: ${record["hostname"]}, namespace: ${record["namespace"]}, agent: ${record["agent"]}, method: ${record["method"]}, path: ${record["path"]}, log: ${record["log"]}
 <filter app.error>
 
   @type record_transformer
-  # add a description of the error to a new key named 'content', and a 'username' required by Discord
-  enable_ruby
+  enable_ruby true
+  auto_typecast true
   <record>
-    content ${record["product"]} - HTTP response status code ${record["code"]}, hostname: ${record["hostname"]}, namespace: ${record["namespace"]}, agent: ${record["agent"]}, method: ${record["method"]}, path: ${record["path"]}
-    username ${record["product"].upcase} (${record["namespace"]}) Error Alert
+    username ${record["product"].upcase} (${record["namespace"]}) Alert
+    content ```json\r${record.to_json}\r```
```

Comment:
Curious as to why this is `\r` instead of `\n` or `\r\n`. It's working at least though.

Reply:
good point. had to refresh my understanding of this. `\n` is a better choice. I've confirmed this by manually updating the config map and it's working.